TelemetryDesignDoc.md

```diff
@@ -0,0 +1,327 @@
+* ShuffleBoard is clearly superior to Glass as an operating dashboard, and has many features (robot-code-driven tabs and layouts, data recording and playback) that make it appealing as the "default" robot dashboard.  However, it is already in danger of slipping into the "legacy code" state of the above two tools due to the migration of developer effort to Glass.  This could eventually cost teams if support for the dashboard becomes difficult to sustain while no other dashboard option offers a similar featureset.
+* Glass is a *fantastic* tool for real-time plotting and signal analysis, but currently lacks a robust featureset to be usable as a dashboard.
+
+Under the current ecosystem, effective users will likely want to make use of both Shuffleboard and Glass, while ignoring the LabView dashboard and SmartDashboard.  However, the documentation and API shapes do not necessarily lead users to this conclusion.  Moreover, developer effort is difficult to split between two tools, but the current distribution of features between tools demands it.
+
+### Shuffleboard performance/maintenance status
+
+Shuffleboard is, essentially, stable abandonware.  However, it has a number of known performance issues and complaints about stability, resource use, and general user experience remain frequent year-after-year.  Given the lack of JavaFX developer impetus/availability/expertise on the WPILib team, and the irrelevance of the platform to the current software "zeitgeist," this is a major stumbling block so long as Shuffleboard's core value-add features (user-code-configurable tab/layout telemetry/dashboard viewer) remain non-duplicated within the ecosystem.
+
+### Overloading of the Sendable implementation
+
+On the surface, the `Sendable` interface offers a powerful abstraction to teams who wish to log their state declaratively rather than imperatively.  However, behind-the-scenes `Sendable` pulls double-duty as the backbone of the `LiveWindow` implementation, which is responsible for auto-populating a list of sensors and actuators for easy debugging.
+
+The `LiveWindow` design philosophy holds that all sensor and actuator objects in code should be accessible via the dashboard while the robot is in Test Mode, *regardless of what the user has done in their robot code.*  While this is a nice idea in theory, it places *huge* constraints on the handling of the `Sendable` interface which often interfere with the latter's use as a telemetry handler:
+
+* While writes to the dashboard work as expected, reads from the dashboard (i.e. calls to the bound setters) are suspended *except* in LiveWindow mode.  This is not documented anywhere on the class, and is extremely inconvenient.
+* The API includes an `isActuator` flag which seemingly serves no purpose other than to disable certain widgets on ShuffleBoard (it does not interact with the other dashboards).
```

Comment:
This would make sense if we want to keep the feature; I'm not averse to that, but I'd rather focus development effort on a new API entirely at this point.  Still might be worth fixing even if it gets deprecated, though?

Reply:
As for the first point, isn't that how SendableChoosers work? and they work not only when LW is enabled....

Reply:
SendableChooser is hardcoded, afaik.  The `Sendable` interface does not ever call the consumers except in test mode.